tests/integration_tests/test_pytorch_distributed.py

```diff
@@ -334,3 +334,38 @@ def test_distributions(storage_mode: str) -> None:
         assert distributions["i"] == optuna.distributions.IntDistribution(0, 1)
         assert distributions["il"] == optuna.distributions.IntDistribution(1, 128, log=True)
         assert distributions["c"] == optuna.distributions.CategoricalDistribution(("a", "b", "c"))
+
+
+@pytest.mark.filterwarnings("ignore::optuna.exceptions.ExperimentalWarning")
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+@pytest.mark.timeout(15, method="signal")
```

Comment:
In my environment, `pytest-timeout` did not terminate the MPI processes as expected.
Could you check if it terminate the processes if you add following lines to `test_updates_state`?

```
       if dist.get_rank() == 0:
            trial.suggest_float("f", 0, 1)
```

Reply:
The test finished after 30 minutes. The `torch.distributed` has the [timeout functionality](https://pytorch.org/docs/stable/distributed.html#torch.distributed.init_process_group), and we do not need to introduce `pytest-timeout`.

```console
tests/integration_tests/test_pytorch_distributed.py:347: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
optuna/_experimental.py:90: in wrapped_init
    _original_init(self, *args, **kwargs)
optuna/integration/pytorch_distributed.py:116: in __init__
    self._number = self._broadcast(getattr(self._delegate, "number", None))
optuna/integration/pytorch_distributed.py:286: in _broadcast
    dist.broadcast(size_buffer, src=0)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

tensor = tensor([5], dtype=torch.int32), src = 0, group = None, async_op = False

    def broadcast(tensor, src, group=None, async_op=False):
        """
        Broadcasts the tensor to the whole group.
    
        ``tensor`` must have the same number of elements in all processes
        participating in the collective.
    
        Args:
            tensor (Tensor): Data to be sent if ``src`` is the rank of current
                process, and tensor to be used to save received data otherwise.
            src (int): Source rank.
            group (ProcessGroup, optional): The process group to work on. If None,
                the default process group will be used.
            async_op (bool, optional): Whether this op should be an async op
    
        Returns:
            Async work handle, if async_op is set to True.
            None, if not async_op or if not part of the group
    
        """
        _check_single_tensor(tensor, "tensor")
        if _rank_not_in_group(group):
            return
    
        opts = BroadcastOptions()
        opts.rootRank = src
        opts.rootTensor = 0
    
        if group is None or group is GroupMember.WORLD:
            default_pg = _get_default_group()
            work = default_pg.broadcast([tensor], opts)
        else:
            group_src_rank = _get_group_rank(group, src)
            opts.rootRank = group_src_rank
            work = group.broadcast([tensor], opts)
        if async_op:
            return work
        else:
>           work.wait()
E           RuntimeError: Application timeout caused pair closure

.venv/lib/python3.8/site-packages/torch/distributed/distributed_c10d.py:1167: RuntimeError
```

Reply:
Fixed by simply lowering the distributed process group default timeout.